sympy/integrals/meijerint.py

```diff
@@ -1810,21 +1813,28 @@ def meijerint_definite(f, x, a, b):
         _debug('Integrand has Singularity Function terms - giving up.')
         return None
 
+    # Check if the limits are already symified or not
+    # is_eq can only be used with sympified objects
+    if not hasattr(a, '__sympy__'):
+        a = _sympify(a)
+    if not hasattr(b, '__sympy__'):
+        b = _sympify(b)
+
     f_, x_, a_, b_ = f, x, a, b
 
     # Let's use a dummy in case any of the boundaries has x.
     d = Dummy('x')
     f = f.subs(x, d)
     x = d
 
-    if a == b:
+    if is_eq(a, b):
         return (S.Zero, True)
 
     results = []
-    if a is -oo and b is not oo:
+    if is_eq(a, S.NegativeInfinity) and not is_eq(b, S.Infinity):
```

Comment:
`is_eq` returns three-valued bool (True, False or None). You can not use plain `not` with this because `not None` gives True.